File: go_app/internal/ui/rapid_diffs/streaming_error.go
package rapid_diffs

import (
	"fmt"
)

// StreamingErrorComponent represents a streaming error component
type StreamingErrorComponent struct {
	Message string
}

// NewStreamingErrorComponent creates a new streaming error component
func NewStreamingErrorComponent(message string) *StreamingErrorComponent {
	return &StreamingErrorComponent{
		Message: message,
	}
}

// Render generates the HTML for the streaming error component
func (s *StreamingErrorComponent) Render() string {
	return fmt.Sprintf(`<streaming-error message="%s"></streaming-error>`, s.Message)
}

File: rust_app/src/components/rapid_diffs/streaming_error.rs
use serde::{Deserialize, Serialize};

/// Renders a streaming error component
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct StreamingErrorComponent {
    /// The error message
    message: String,
}

impl StreamingErrorComponent {
    /// Create a new streaming error component
    pub fn new(message: String) -> Self {
        Self { message }
    }

    /// Render the streaming error component
    pub fn render(&self) -> String {
        format!(
            r#"<div class="streaming-error" data-message="{}"></div>"#,
            self.message
        )
    }
} 